src/services/getTokenFromExchange.ts

```diff
@@ -0,0 +1,108 @@
+import { Erc20Api, TokenDetails, TokenList, ExchangeApi } from 'types'
+import { getImageUrl, log, getToken } from 'utils'
+import { getErc20Info } from './getErc20Info'
+
+interface TokenFromErc20Params {
+  tokenAddress: string
+  tokenId: number
+  erc20Api: Erc20Api
+}
+
+async function getTokenFromErc20(params: TokenFromErc20Params): Promise<TokenDetails | null> {
+  const { tokenAddress, tokenId, erc20Api } = params
+
+  // Get base info from the ERC20 contract
+  const erc20Info = await getErc20Info({ address: tokenAddress, erc20Api })
+  if (!erc20Info) {
+    log('Could not get details for token token (%s)', tokenAddress)
+    return null
+  }
+
+  const token = {
+    ...erc20Info,
+    id: tokenId,
+    address: tokenAddress,
+    image: getImageUrl(tokenAddress),
+  }
+
+  // TODO: add to tokenList?
+
+  return token
+}
+
+interface BaseParams {
+  networkId: number
```

Comment:
this might change too often, but not the APIs. The factory is only initialized once.

Reply:
The networkId is part of the external function parameters as well, so it won't be cached in the factory.

I need networkId for searching the local list of tokens before doing any external call.

The issue is that we keep a list of tokens for each network and the network id is required to use `tokenListApi.getTokens`.

Any ideas to make `tokenListApi.getTokens` network aware/agnostic?

Reply:
I guess now that I created the factories the code makes more sense.

Reply:
What I mean, is that it should be like this imo:

```js
async function getTokenFromErc20Factory({  
  erc20Api,
}: TokenFromErc20Params): Promise<TokenDetails | null> {
    
   return ({ tokenAddress: string, tokenId: number } ) = {
        .... // logic
   }
}
```

Reply:
Forget my last comment, I see that that's what you did. I saw `getTokenFromErc20` and I got confused cause thought it was what it was exporting (and I saw mixed the API with the tokenId...)